### ISSUE TITLE
Bayonetta 3 - 01004A4010FEA000

### ISSUE THREAD

**Issue by EmulationFanatic** [Ryujinx/Ryujinx-Games-List#4129]
#### Graphics Backend : Vulkan & OpenGL
On OpenGL, the game boots but has some serious graphical deficiencies starting at the title screen and continuing on in-game, making it unplayable.
On Vulkan, the game loads and goes in-game smoothly, but suffers from several issues:

- Missing lighting/textures at the title screen.
- "god rays" during gameplay - bright plasma-like rays covering a portion to nearly all of the screen. Some have reported that downgrading Nvidia drivers to 516.40 resolves these anomalies.
- Photo Mode is essentially useless because the 'Capture' button is not mappable/implemented.
- Cutscenes frequently desync the audio & video, with lower end platforms suffering worse desync than higher end platforms.
- During some combat scenes, the game seems to switch into a low resolution with a green tint on everything. While this was happening, multiple console errors were spammed:
```
00:08:49.417 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Ram is not implemented.
00:08:49.417 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Sam is not implemented.
00:08:49.417 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Ram is not implemented.
00:08:49.417 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Sam is not implemented.
00:08:49.417 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Ram is not implemented.
00:08:49.417 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Sam is not implemented.
```

There may be other visual bugs I didn't encounter or remember after a few gameplay sessions, but feel free to add any others.
I did not experience any crashes, but those with less VRAM may run into Out of Memory errors. Enabling Texture Recompression seems to fix this. 
Tl;dr: The game can be played through on Vulkan, with some minor to moderate issues

#### Screenshots :

Title screen too dark:

![2022-10-28 07_44_54-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198620916-89306332-9bc2-413c-817a-afbe6254d140.png)

God rays:
![2022-10-28 07_47_00-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198621749-01a09126-3a5a-4ced-869e-50f329867821.png)

![2022-10-28 07_49_13-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198626441-080c2766-24cf-48f6-8ea7-97347c81599d.png)
![2022-10-28 07_51_18-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198626475-729b1927-c091-4026-ba19-f871c114a7fd.png)
![2022-10-28 07_52_24-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198626503-2657d564-3d22-4454-b61b-852b896fa6d1.png)
![2022-10-28 07_52_57-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198626523-db2a56fb-6276-4b9b-9f34-91bc055489f6.png)
![2022-10-28 07_53_59-Ryujinx 1 1 330 - Bayonetta 3 v1 1 0 (01004A4010FEA000) (64-bit)](https://user-images.githubusercontent.com/62343878/198626550-30551584-3f53-4c96-995b-02ff5e1531d8.png)

**Comment by PhiZero** [comment]
Nvidia driver 516.94 (or 516.40) indeed fix the god rays.

https://user-images.githubusercontent.com/7350221/198684004-2acf6c32-011e-4406-bd60-9dd2223016e1.mp4

**Comment by serpentiem** [comment]
The game crashes in chapter 1 during the second cutscene directly after she says "What's that?" Which is kinda funny I guess.

CPU: AMD Ryzen 7 2700X
GPU: AMD Radeon RX 580
GPU Driver Version: 22.10.3
RAM: 16GB
System Version: 15.0.0
Game Version: 1.1.0

Latest emu version (1.1.330) and default settings.

Up until the crash the game's very playable - a few texture issues here and there.

Nothing on my end indicates that I'm running out of RAM and or VRAM - I even set my page file to 32GB just in case. This seems to be a bug.

Here's the log:
```
00:01:33.694 |I| .NET ThreadPool Worker Ptc Save: Saved Profiling Info (size: 76244 bytes, profiled functions: 22424).
00:01:56.766 |S| HLE.OsThread.16 ServiceAm SetIdleTimeDetectionExtension: Stubbed. {idleTimeDetectionExtension: 0}
00:01:56.778 |S| HLE.OsThread.16 ServiceAm SetIdleTimeDetectionExtension: Stubbed. {idleTimeDetectionExtension: 1}
00:02:24.357 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:02:41.783 |S| HLE.OsThread.16 ServiceAm SetIdleTimeDetectionExtension: Stubbed. {idleTimeDetectionExtension: 0}
00:02:41.808 |S| HLE.OsThread.16 ServiceAm SetIdleTimeDetectionExtension: Stubbed. {idleTimeDetectionExtension: 1}
00:03:03.705 |I| .NET ThreadPool Worker Ptc Save: Saved Profiling Info (size: 76244 bytes, profiled functions: 22424).
00:03:04.133 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:03:12.547 |W| GPU.MainThread Gpu Log: Shader translator: Shader instruction Votevtg is not implemented.
00:03:12.719 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Ryujinx.Graphics.Vulkan.Shader+<>c__DisplayClass17_0.<.ctor>b__0()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread, System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef, System.Threading.Thread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
```

**Comment by GamerzHell9137** [comment]
Random white square can appear on the screen in game.
Also 317 Nvidia driver is the latest one that doesn't show the godray issue.

https://user-images.githubusercontent.com/2002038/198751068-54fd397f-b1b7-46d9-8af4-784794f42f3d.mp4

**Comment by MetrosexualGarbodor** [comment]
As of 1.1.332, it no longer crashes on AMD GPUs.

**Comment by emmandyar** [comment]
1.1.334 still crashes on nvidia, vulkan device out of memory (6gb gtx 1660 super). If I play at native (currently at 2x) it takes longer but eventually crashes with the same error

**Comment by MetrosexualGarbodor** [comment]
@emmandyar you have to use texture recompression if you're getting device out of memory errors.

**Comment by emmandyar** [comment]
Yess! That did it!

Thanks!!

**Comment by GeorgSci** [comment]
My vulkan game crashes

**Comment by 0Lambda** [comment]
1.1.335, 517 Nvidia driver, gpu 3080ti, the game is running at 15-17 fps. Turning on/off vsync, dram size and other parameters has no effect.

**Comment by MetrosexualGarbodor** [comment]
@0Lambda join discord for help

**Comment by pockercv94** [comment]
Yo-yo weapon give you a spider form, And let you to wall crawl. This doesn`t work on emulator.

**Comment by marchenko-ia12** [comment]
I can confirm Godrays and other similar bugs on AMD 6800xt with 22.10.1 driver. The problem is not only on Nvidia's side

**Comment by contra223** [comment]
game is stuck on chapter 1 infinite loading screen u can walk and enter practice mode but it never finish loading could load fine on prologue  

ryujinx 1.1.335
nivida 2080 super driver 526.47
ryzen 3700x
32gb ram
win 11



![bug](https://user-images.githubusercontent.com/117041386/198899557-00bf3242-0ce8-490e-9666-fd91335bf316.jpg)

**Comment by tedios** [comment]
Finished the game fully, except the hidden/chapter book as it's not loading via Ryujinx - only issue is the audio desync that sometimes occurs during cutscenes. 

No crashes since installing Nvidia's 516.40 drivers and using the hacks

Ryujinx version: 1.1.335
Game version: 1.1.0
Cheats/Mods: All disabled

System Specs:
OS: Windows 10 Pro
CPU: AMD Ryzen 9 5900X (12) @ 3.7GHz
Memory: 32 GB
GPU: Nvidia RTX 3080 12GB - 516.40 drivers

Ryujinx Settings:
System section
Vsync: ON
PPTC: Enabled
FS Integrity Checks: Enabled
Audio Backend: SDL2
Memory Manager Mode: Host (fast)
Expand DRAM Size to 6GB
Ignore Missing services

Graphics section
Graphics Backend Multithreading: Auto
Graphics Backend: Vulkan
Shader Cache: Enabled
Resolution Scale: Native

**Comment by Javy108** [comment]
> 

Do you have shaders cache?

**Comment by PasswordsAreUseless** [comment]
Do we know a way to fix the Chapter 1 loading bug?
Gotta say it's a massive pain in the ass to hit a brick wall only 20 minutes into the game

**Comment by MetrosexualGarbodor** [comment]
@PasswordsAreUseless if you get infinite loading your game file is broken and you need to dump it again.

**Comment by Bjorn29512** [comment]
**DESCRIPTION :** 
Game suffers from Incorrect/Partial Fog rendering.

**SPECS :** 

    12th Gen Intel(R) Core(TM) i5-12400F | 12 Threads | 31.82 GiB RAM | TSC: 2.496GHz | AVX-512+ | FMA3
    Operating system: Windows, Major: 10, Minor: 0, Build: 22000, Service Pack: none, Compatibility mode: 0
    GPU: 'NVIDIA GeForce GTX 1660 SUPER' running on driver 526.47.0.0

**MEDIA FILES SHOWCASING THE ISSUE :** 

Bugged : 
![image](https://user-images.githubusercontent.com/110204265/199266463-cad3a15b-a96d-4c73-85c3-25d14655f9c0.png)

Switch : 
![image](https://user-images.githubusercontent.com/110204265/199266675-942cdaab-a1ff-430e-a60c-677b72a9a34a.png)

**REPRODUCTION STEPS AND SAVE FILE :** 
- Press continue and walk towards the fog.
[0000000000000002.zip](https://github.com/Ryujinx/Ryujinx-Games-List/files/9911619/0000000000000002.zip)

**LOG FILE :**
[Ryujinx_1.1.335_2022-11-01_20-49-59.log](https://github.com/Ryujinx/Ryujinx-Games-List/files/9911627/Ryujinx_1.1.335_2022-11-01_20-49-59.log)

**Comment by Infinit2003** [comment]
Did anyone have a problem that the game just crashes after 1 mini boss ( This is when Bayonetta mastered 1 demon ) ? MY SYSTEM
NVIDIA GeForce GTX 1060 6GB
Driver version 517.48
Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
15.94 GB RAM
1920 x 1080, 60Hz

**Comment by MetrosexualGarbodor** [comment]
@Infinit2003 Join discord for support, this thread is for issue reports. https://discord.gg/ryujinx

**Comment by MetrosexualGarbodor** [comment]
@Kismetic1 Join discord for support, this thread is for issue reports. https://discord.gg/ryujinx

**Comment by Pixxel123** [comment]
Visual pixelation artifacts appear on AMD GPU's, in this case using a Steam Deck. This does not appear in Nvidia GPU's (GTX 970). 

Attached are screenshots from both hardware, both using Ryujinx 1.1.138, Vulkan graphics backend for both, both at native resolution at handheld. Texture Recompression is on for both.

AMD hardware:

![image](https://user-images.githubusercontent.com/36304206/200186616-9cf254f5-45f5-4c12-9ff1-1dc78d579c9f.png)

NVIDIA hardware:

![image](https://user-images.githubusercontent.com/36304206/200186693-11704b5d-8db4-4821-be01-d2a3ebbd3328.png)

**Comment by Bjorn29512** [comment]
Attaching some more examples of incorrect dark lighting instances in the game 
![image](https://user-images.githubusercontent.com/110204265/201996874-56c1e515-5204-4054-8746-72cfb70f027b.png)
![Ryujinx Nvidia Profile Screenshot 2022 11 15 - 23 56 01 66](https://user-images.githubusercontent.com/110204265/201996891-f900d265-be75-4d83-a527-f9d6c693812a.png)
![Ryujinx Nvidia Profile Screenshot 2022 11 15 - 23 56 08 72](https://user-images.githubusercontent.com/110204265/201996897-ced82195-3ab9-4bd5-b5f9-a70c2a1f94bd.png)

**Comment by markmcclements** [comment]
Any advice on how to fix the god rays if your on a 4090 and cant rollback to the old drivers as the card is only supported from 522.25 onwards?

**Comment by ggspita** [comment]
![image](https://user-images.githubusercontent.com/119282451/204206101-c316c198-53a5-4cb6-aed5-a65de9432820.png)
fui tentar jogar porem da esse erro de textura

**Comment by gdkchan** [comment]
Usa Vulkan.

**Comment by markmcclements** [comment]
Thanks for the advice but I am already using vulkan

**Comment by VenomApocalypse** [comment]
Crashes during the cutscene where another Bayonetta fuses with the burning spider. Any known fix for this?

**Comment by Simptaz** [comment]
I've been experiencing crashes during chapter 8 when you enter the underground temple. I fight the boss and it repeatedly crashed: it started when I tried to open a chest on the second floor, then in the room with the white seeds where you need to make it rain to solve the puzzle door (I was able to make it past this room by staying close to the entrance and performing the action from there) then it happened as soon as the battle in the following room starts.

2080 Super, 10th Core i7 10700k

I got VSync, PPTC, and FS Integrity Checks enabled
Memory Manager Mode is set to Host and I have Hacks "Use alternative memory layout" and "Ignore Missing Services"

Had Graphics Backend Multithreading on "Auto" and then switched to "On" with no change.

Using Vulkan and have Shader Cache and Macro HLE clicked and Anisotropic Filtering set to Auto.

**Comment by thurnissilva** [comment]
audiosynch was fixed on 1.1.415

**Comment by Aphonx** [comment]
A Advice, my game crash in chapter 8 in egipty fase , anyone have this same trouble ?

**Comment by anxie-tea** [comment]
So after pressing "Press Any Key/Button to Start", my game crashes and I don't have a clue as to why
AMD Radeon 6700 XT, AMD Ryzen 9 5900X
![ryujinx error](https://user-images.githubusercontent.com/126358796/221380056-5e33a7d4-18f9-451b-a791-b5759eb39c44.png)

**Comment by MetrosexualGarbodor** [comment]
@anxie-tea your game file is broken, dump it again. And please use discord for help questions.

**Comment by gyuchann** [comment]
![bayonetta3 crash](https://user-images.githubusercontent.com/132113729/235277156-3874650f-6810-443e-8949-21cd31426e7d.png)

Crash after loading Chapter 3

CPU : Ryzen 5600X
GPU : Intel Arc A770 (16GB)
RAM : Corsair DDR4 3600 32GB
[Ryujinx_1.1.738_2023-04-29_10-12-25.log](https://github.com/Ryujinx/Ryujinx-Games-List/files/11358079/Ryujinx_1.1.738_2023-04-29_10-12-25.log)

**Comment by Tintinkung13** [comment]
how status for this game?

**Comment by gamobigz3** [comment]
> how status for this game?

It's playable to Chapter 8. I keep getting crashes. Just after the Baal Mini game. As I exit the Pyramid.

![ryujinx_capture_2023-06-24_09-49-44](https://github.com/Ryujinx/Ryujinx-Games-List/assets/137640106/1e8e1d20-e8bd-4a97-a45a-ed27d37910c6)

As a side note. You will get occasional crashes even before chapter 8. it's just a question of whether you are patient enough to keep trying to play through the part where it crashes.

My Hardware Specs are in this log: 
[Ryujinx_1.1.907_2023-06-24_09-51-02.log](https://github.com/Ryujinx/Ryujinx-Games-List/files/11857625/Ryujinx_1.1.907_2023-06-24_09-51-02.log)

Rujinx Version: 1.1.907
Game version: Bayonetta 3 v1.2.0
Firmware/System Version: 16.0.3
CPU : Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz ; 8 logical
GPU : NVIDIA GeForce GTX 1060 (Vulkan v1.3.224, Driver v526.47.0.0)
RAM :  Total 16340 MiB ; Available 12731 MiB


My Ryujinx interface.
![computer hardware specs 6-24-2023](https://github.com/Ryujinx/Ryujinx-Games-List/assets/137640106/123fa947-0e1b-491b-9456-0cc21d6a8665)

**Comment by Aphonx** [comment]
I have trouble in this part too. After the cutscene ryujinx crash immediately.

**Comment by OExSTir** [comment]
[Ryujinx_1.1.1089_2023-11-18_07-57-50.log](https://github.com/Ryujinx/Ryujinx-Games-List/files/13399484/Ryujinx_1.1.1089_2023-11-18_07-57-50.log)
hi im new to this im having trouble running the game i did the dump like the discord told me even switched between openGL and vulkan plz help

**Comment by HibernantBear** [comment]
> Attaching some more examples of incorrect dark lighting instances in the game ![image](https://user-images.githubusercontent.com/110204265/201996874-56c1e515-5204-4054-8746-72cfb70f027b.png) ![Ryujinx Nvidia Profile Screenshot 2022 11 15 - 23 56 01 66](https://user-images.githubusercontent.com/110204265/201996891-f900d265-be75-4d83-a527-f9d6c693812a.png) ![Ryujinx Nvidia Profile Screenshot 2022 11 15 - 23 56 08 72](https://user-images.githubusercontent.com/110204265/201996897-ced82195-3ab9-4bd5-b5f9-a70c2a1f94bd.png) ![Ryujinx Nvidia Profile Screenshot 2022 11 17 - 00 24 50 22](https://user-images.githubusercontent.com/110204265/202267810-809a1614-8078-4261-8552-160b89afc122.png)

is there any plan for fixing this? switch performs foor for this game, so if this could be fixed we can have fun with Bayonetta on Ryujinx :p
thanks alot

**Comment by joseeeewhdkjdhdkj** [comment]
where is the key? to instal

**Comment by zhm86** [comment]
> > Attaching some more examples of incorrect dark lighting instances in the game ![image](https://user-images.githubusercontent.com/110204265/201996874-56c1e515-5204-4054-8746-72cfb70f027b.png) ![Ryujinx Nvidia Profile Screenshot 2022 11 15 - 23 56 01 66](https://user-images.githubusercontent.com/110204265/201996891-f900d265-be75-4d83-a527-f9d6c693812a.png) ![Ryujinx Nvidia Profile Screenshot 2022 11 15 - 23 56 08 72](https://user-images.githubusercontent.com/110204265/201996897-ced82195-3ab9-4bd5-b5f9-a70c2a1f94bd.png) ![Ryujinx Nvidia Profile Screenshot 2022 11 17 - 00 24 50 22](https://user-images.githubusercontent.com/110204265/202267810-809a1614-8078-4261-8552-160b89afc122.png)
> 
> is there any plan for fixing this? switch performs foor for this game, so if this could be fixed we can have fun with Bayonetta on Ryujinx :p thanks alot

![261633821-3f26de1b-fd03-4912-a34a-838a0855dc7c](https://github.com/user-attachments/assets/bb006154-8519-4008-ba83-288fcb020e98)
![261633854-4d72c4fe-be4d-42b2-91c3-56f9ad0d2ca4](https://github.com/user-attachments/assets/f8239570-8ec0-45d2-8742-66655a8caa99)

Same here. It may be a game specific bug, or something wrong with the ASTC texture decoder?

**Comment by joseeeewhdkjdhdkj** [comment]
Hello, well, look, do you know where the bayonetta 3 can be installed?
I always install it in one place, it gives me an error, and well, if
you have the link on how to download it, would you help me?

**Comment by DniweTamp** [comment]
Regarding the broken lighting, it appears there's a very small block on the top left of the screen, which does render it right. It can change size over time in cutscenes.


https://github.com/user-attachments/assets/8bb1194f-449a-4460-933f-da553a947b0b

**Comment by DniweTamp** [comment]
Probably unrelated to broken lighting, but auto exposure is also not working

How it is currently on Ryujinx:

https://github.com/user-attachments/assets/2e894190-c2e8-4f17-8531-49b19d54a765

Expected behavior (notice how the room interior only gets brighter shortly after you look away from the outdoor area):

https://github.com/user-attachments/assets/d6076cfd-79eb-4f5e-879e-824a5e2e6956
